wandb/sdk/launch/docker.py

```diff
@@ -120,40 +133,58 @@ def build_docker_image(
         base_url = "http://host.docker.internal:9002"
     else:
         base_url = api.settings("base_url")
-    image_uri = _get_docker_image_uri(name=project.name, work_dir=project.project_dir)
+    image_name = "wandb_launch_{}".format(launch_project.run_id)
+    image_uri = _get_docker_image_uri(
+        name=image_name, work_dir=launch_project.project_dir
+    )
     copy_code_line = ""
     workdir_line = ""
+    copy_config_line = ""
+    workdir = os.path.join("/home/", getpass.getuser())
+    if launch_project.override_config:
+        copy_config_line = "COPY {}/{} {}\n".format(
+            _PROJECT_TAR_ARCHIVE_NAME, _project_spec.DEFAULT_CONFIG_PATH, workdir
+        )
     if copy_code:
-        workdir = os.path.join("/home/", getpass.getuser())
         copy_code_line = "COPY {}/ {}\n".format(_PROJECT_TAR_ARCHIVE_NAME, workdir)
         workdir_line = "WORKDIR {}\n".format(workdir)
-    wandb_project = project.target_project
-    wandb_entity = project.target_entity
+    name_line = ""
+    if launch_project.name:
+        name_line = "ENV WANDB_NAME={wandb_name}\n"
+    project = launch_project.target_project
+    entity = launch_project.target_entity
+
     dockerfile = (
         "FROM {imagename}\n"
+        "{copy_config_line}"
         "{copy_code_line}"
         "{workdir_line}"
+        "{name_line}"
         "ENV WANDB_BASE_URL={base_url}\n"
         "ENV WANDB_API_KEY={api_key}\n"
-        "ENV WANDB_PROJECT={wandb_project}\n"
-        "ENV WANDB_ENTITY={wandb_entity}\n"
-        "ENV WANDB_NAME={wandb_name}\n"
+        "ENV WANDB_PROJECT={project}\n"
+        "ENV WANDB_ENTITY={entity}\n"
         "ENV WANDB_LAUNCH=True\n"
         "ENV WANDB_LAUNCH_CONFIG_PATH={config_path}\n"
         "ENV WANDB_RUN_ID={run_id}\n"
```

Comment:
is the change to move this to the run command in a different pr?

Reply:
Ah forgot to merge that one in.